### PR TITLE
[symbols] stop fetching symbols when loading

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

We were previously fetching symbols again when we were loading because hasSymbols ignores loading.

### CODE REVIEW

File: src/actions/ast.js [hunk 1]
```diff
@@ -92,7 +87,7 @@ export function setOutOfScopeLocations() {
 
     let locations = null;
     if (location.line && source && !source.isWasm && isPaused(getState())) {
-      locations = await findOutOfScopeLocations(
+      locations = await parser.findOutOfScopeLocations(
```
Comment (jasonLaster) [inline src/actions/ast.js]: adding some more specificity so that it is clear when we use a selector vs worker

File: src/actions/ast.js [hunk 2]
```diff
@@ -61,14 +56,14 @@ export function setSymbols(sourceId: SourceId) {
   return async ({ dispatch, getState, sourceMaps }: ThunkArgs) => {
     const source = getSourceFromId(getState(), sourceId);
 
-    if (source.isWasm || hasSymbols(getState(), source) || !isLoaded(source)) {
+    if (source.isWasm || !isLoaded(source) || getSymbols(getState(), source)) {
```
Comment (jasonLaster) [inline src/actions/ast.js]: `hasSymbols` was false when loading

**Comment by bomsy** [comment]
hmm ... these tests shouldn't fail with these changes

File: src/actions/ast.js [hunk 3]
```diff
@@ -61,15 +56,17 @@ export function setSymbols(sourceId: SourceId) {
...
-      sourceId,
-      [PROMISE]: getSymbols(sourceId)
-    });
+    if (!getSymbols(getState(), source)) {
+      await dispatch({
+        type: "SET_SYMBOLS",
+        sourceId,
+        [PROMISE]: parser.getSymbols(sourceId)
+      });
+    }
```
Comment (bomsy) [inline src/actions/ast.js]: This fixes the  failing tests, since we still want o fetch the extra content even if we don't reset the symbols.

**Comment by jasonLaster** [comment]
@bomsy i reverted some of your changes to see if it would pass after a rebase. I thought this should be simple :/

**Comment by bomsy** [comment]
@jasonLaster that's a good idea. didn't try that first.

It seems though like the `loading` check in `hasSymbols` was there to fix this `extra` property issue in the failing test, which is weird and a bit confusing. Might need a refactor. Would love to pair when you have a chance.

**Pull request merged**